Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.js

```diff
@@ -67,24 +74,107 @@ gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawStar = function(x1, y1,
     this._graphics.endFill();
 };
 
-// These functions are disabled as the way they work is unsure. See https://github.com/4ian/GDevelop/pull/1256
-/*
-gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawArc = function(x1, y1, radius, startAngle, endAngle, anticlockwise) {
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawArc = function(x1, y1, radius, startAngle, endAngle, anticlockwise, closePath) {
     this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(  x1 + radius * Math.cos(gdjs.toRad(startAngle)), y1 + radius * Math.sin(gdjs.toRad(startAngle)));
     this._graphics.arc(x1, y1, radius, gdjs.toRad(startAngle), gdjs.toRad(endAngle), anticlockwise ? true : false);
-    //this._graphics.closePath();
+    if(closePath){
+        this._graphics.closePath();
+    }
     this._graphics.endFill();
 };
-    
-gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawArcTo = function(x1, y1, x2, y2, radius) {
+
+/*    
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawArcTo = function(x1, y1, x2, y2, radius, closePath) {
     this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
     this._graphics.moveTo(200,200);
     this._graphics.arcTo(x1, y1, x2, y2, radius);
-    //this._graphics.closePath();
+    if(closePath){
+        this._graphics.closePath();
+    }
+    this._graphics.endFill();
+};
+*/
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawBezierCurve = function(x1, y1, cpX, cpY, cpX2, cpY2, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.bezierCurveTo(cpX, cpY, cpX2, cpY2, x2, y2);  
     this._graphics.endFill();
 };
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.drawQuadraticCurve = function(x1, y1, cpX, cpY, x2, y2) {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    this._graphics.moveTo(x1, y1);
+    this._graphics.quadraticCurveTo(cpX, cpY, x2, y2);
+    this._graphics.endFill();
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.beginFillPath = function() {
+    this._graphics.beginFill(this._object._fillColor, this._object._fillOpacity / 255);
+    /*if(!this.hasBeginFillPath){
+        this.beginFillPath();
+        this._graphics.haveBeginFillPath = true;
+    }*/
+};
+
+gdjs.ShapePainterRuntimeObjectPixiRenderer.prototype.endFillPath = function() {
+   /* if(!this.hasBeginFillPath){
```

Comment:
The commented code is not needed? If we call this._graphics.endFill without having starting a path, there is no crash or weird result?

Reply:
hasBeginFillPath / hasBegunFillPath
Is not used anymore i delete this useless function and all comments associate.